config_gen.py

```diff
@@ -162,6 +163,7 @@ def fake_build(project_dir, c_build_log_path, cxx_build_log_path, verbose, make_
     env_config = env.copy()
     env_config["YCM_CONFIG_GEN_CC_PASSTHROUGH"] = cc
     env_config["YCM_CONFIG_GEN_CXX_PASSTHROUGH"] = cxx
+    env_config["QMAKESPEC"] = "linux-clang"
```

Comment:
This should be performed within the if-case for qmake. (Similar to how the cmake case sets `cwd`.)